### PR TITLE
Extract shared config classes for DPO training scripts

### PR DESCRIPTION
**Pull request by finbarrtimbers**
Create dpo_config.py with shared configuration dataclasses:
- ExperimentConfig: experiment tracking (exp_name, run_name, seed)
- ModelConfig: model loading configuration
- DPOHyperparamsConfig: DPO-specific hyperparameters
- TrainingConfig, DatasetConfig, LoRAConfig, LoggingConfig
- HubConfig, CheckpointConfig, EvalConfig

Update dpo_tune_cache.py to inherit from these shared configs, reducing code duplication between DPO implementations.

Ran single GPU DPO script ([Beaker](https://beaker.org/ex/01KEZJD4BJ4857P5SXVJZHMC8R)) and multi-node DPO ([Beaker](https://beaker.org/ex/01KEZJQ3EA95YZCSHXXCNCQC4Z)) scripts.

### CODE REVIEW

File: open_instruct/dpo_config.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    packing: bool = False
+    """Whether to use packing/padding-free collation."""
+
+
+@dataclass
+class TrainingConfig:
+    """Configuration for training hyperparameters."""
+
+    num_epochs: int = 2
+    """Total number of training epochs to perform."""
```
[the remark this answers is not in the document]
Comment (finbarrtimbers) [reply open_instruct/dpo_config.py:68]: I have updated all scripts!

File: open_instruct/dpo_tune_cache.py [hunk 2]
```diff
@@ -107,89 +117,22 @@ class FlatArguments:
...
-        default=False,
-        metadata={
-            "help": (
-                "It is an option to create the model as an empty shell, "
-                "then only materialize its parameters when the pretrained weights are loaded. "
-                "set True will benefit LLM loading time and RAM consumption."
-            )
-        },
-    )
     dataset_name: str | None = field(
```
Comment (hamishivi) [inline open_instruct/dpo_tune_cache.py:130]: I guess that you have some olmo-core DPO implementation that's not using these flags? Otherwise confused on why some flags are left behind (e.g. dataset_name).

**Review by hamishivi — APPROVED** [review]
Seems fine, see comment- I think I'm just missing some context? We should probably clean up this script afterwards.
